src/rules/commentFormatRule.ts

```diff
@@ -27,8 +27,23 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class CommentWalker extends Lint.RuleWalker {
+    private tokensToSkipStartEndMap: {[start: number]: number};
+
+    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+        super(sourceFile, options);
+        this.tokensToSkipStartEndMap = {};
+    }
+
     public visitSourceFile(node: ts.SourceFile): void {
+        super.visitSourceFile(node);
         Lint.scanAllTokens(ts.createScanner(ts.ScriptTarget.ES5, false, node.text), (scanner: ts.Scanner) => {
+            var startPos = scanner.getStartPos();
+            if (this.tokensToSkipStartEndMap[startPos] != null) {
+                // tokens to skip are places where the scanner gets confused about what the token is, without the proper context
+                // (specifically, regex, identifiers, and templates). So skip those tokens.
+                scanner.setTextPos(this.tokensToSkipStartEndMap[startPos]);
+                return;
+            }
```

Comment:
There should probably be a newline between the two if blocks here -- but it shouldn't be an else if, the two things are different concepts (the first block is saying "if this is true, continue", which doesn't make sense to group with the next block.)